airflow/kubernetes/pod_launcher_deprecated.py

```diff
@@ -0,0 +1,300 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Launches PODs"""
+import json
+import math
+import time
+from datetime import datetime as dt
+from typing import Optional, Tuple
+
+import pendulum
+import tenacity
+from kubernetes import client, watch
+from kubernetes.client.models.v1_pod import V1Pod
+from kubernetes.client.rest import ApiException
+from kubernetes.stream import stream as kubernetes_stream
+from requests.exceptions import BaseHTTPError
+
+from airflow.exceptions import AirflowException
+from airflow.kubernetes.kube_client import get_kube_client
+from airflow.kubernetes.pod_generator import PodDefaults
+from airflow.settings import pod_mutation_hook
+from airflow.utils.log.logging_mixin import LoggingMixin
+from airflow.utils.state import State
+
+
```

Comment:
Should we also a the deprecation warning directly here (or even move the warning from the old pod_launcher)? This way even if someone by mistake imports PodLauncher/PodStatus directly from there (because IDE will provide this as first-class option), will still get the warning.

Reply:
@potiuk I added the deprecation to the deprecated class. good catch!

Reply:
Yep.. Waiting for push to approve it ;)

Reply:
@potiuk oh oops 🤦 . Pushed!